wgpu/tests/resource_error.rs

```diff
@@ -0,0 +1,61 @@
+use crate::common::{initialize_test, TestParameters};
+
+#[cfg(test)]
+// Run some code in an error scope and assert that validation failed.s
+fn fail(device: &wgpu::Device, callback: &mut dyn FnMut()) {
+    device.push_error_scope(wgpu::ErrorFilter::Validation);
+    callback();
+    assert!(pollster::block_on(device.pop_error_scope()).is_some());
+}
+
+#[test]
+fn bad_buffer() {
+    // Create a buffer with bad parameters and call a few methods.
+    // Validation should fail but there should be not panic.
+    initialize_test(TestParameters::default(), |ctx| {
+        ctx.device.push_error_scope(wgpu::ErrorFilter::Validation);
+        let buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+            label: None,
+            size: 99999999,
+            usage: wgpu::BufferUsages::MAP_READ | wgpu::BufferUsages::STORAGE,
+            mapped_at_creation: false,
+        });
+        assert!(pollster::block_on(ctx.device.pop_error_scope()).is_some());
+
+        fail(&ctx.device, &mut || {
+            buffer.slice(..).map_async(wgpu::MapMode::Write, |_| {})
+        });
+        fail(&ctx.device, &mut || buffer.unmap());
+        fail(&ctx.device, &mut || buffer.destroy());
+        fail(&ctx.device, &mut || buffer.destroy());
+    });
+}
+
+#[test]
+fn bad_texture() {
+    // Create a texture with bad parameters and call a few methods.
+    // Validation should fail but there should be not panic.
+    initialize_test(TestParameters::default(), |ctx| {
+        ctx.device.push_error_scope(wgpu::ErrorFilter::Validation);
```

Comment:
Why can't this block also use `fail`

Reply:
I did not bother with making it possible for fail() to return a value, but if this is to be hoisted, I should probably make an effort.